projects/storefrontlib/src/cms-structure/outlet/provide-outlet.ts

```diff
@@ -0,0 +1,41 @@
+import {
+  APP_INITIALIZER,
+  ComponentFactoryResolver,
+  FactoryProvider,
+  Type,
+} from '@angular/core';
+import { OutletPosition } from './outlet.model';
+import { OutletService } from './outlet.service';
+
+interface ProvideOutletOptions {
+  name: string;
+  position?: OutletPosition;
+  component: Type<any>;
+}
+
+export function provideOutlet(options: ProvideOutletOptions): FactoryProvider {
```

Comment:
did you add the option complex argument for flexiblity in the future?

Reply:
Yes. In my previous (more advanced) implementation, you could pass here `providers: []`, to inject into the outlet component (and all its descendants) additional service.